### PR TITLE
HIVE-21325 : Hive external table replication failed with Permission denied issue.

### PR DESCRIPTION
**Pull request by maheshk114**
…

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/security/authorization/StorageBasedAuthorizationProvider.java [hunk 1]
```diff
@@ -188,6 +191,17 @@ public void authorize(Table table, Privilege[] readRequiredPriv, Privilege[] wri
       checkDeletePermission(path, getConf(), authenticator.getUserName());
     }
 
+    try {
+      String user = SecurityUtils.getUGI().getShortUserName();
```
Comment (sankarh) [inline ql/src/java/org/apache/hadoop/hive/ql/security/authorization/StorageBasedAuthorizationProvider.java]: Can we move it before checkDeletePermission?

File: ql/src/java/org/apache/hadoop/hive/ql/security/authorization/StorageBasedAuthorizationProvider.java [hunk 2]
```diff
@@ -188,6 +191,17 @@ public void authorize(Table table, Privilege[] readRequiredPriv, Privilege[] wri
...
+      if (MetaStoreServerUtils.checkUserHasHostProxyPrivileges(user, getConf(),
+              HiveMetaStore.HMSHandler.getIPAddress())) {
+        LOG.info("Path authentication is skipped for user {} with host proxy privilege." + user);
```
Comment (sankarh) [inline ql/src/java/org/apache/hadoop/hive/ql/security/authorization/StorageBasedAuthorizationProvider.java]: Path "authorization" not "authentication".

Comment (sankarh) [inline ql/src/java/org/apache/hadoop/hive/ql/security/authorization/StorageBasedAuthorizationProvider.java]: Also, can log the path.

File: ql/src/java/org/apache/hadoop/hive/ql/security/authorization/StorageBasedAuthorizationProvider.java (hunk 1 above)
Comment (maheshk114) [reply ql/src/java/org/apache/hadoop/hive/ql/security/authorization/StorageBasedAuthorizationProvider.java]: if required will be done alter

File: ql/src/java/org/apache/hadoop/hive/ql/security/authorization/StorageBasedAuthorizationProvider.java [hunk 3]
```diff
@@ -188,6 +191,17 @@ public void authorize(Table table, Privilege[] readRequiredPriv, Privilege[] wri
...
+      }
+    } catch (Exception ex) {
+      // if can not decide on the proxy privilege status, then proceed with authorization check.
+      LOG.error("Cannot obtain username to check for host proxy privilege", ex);
```
[the remark this answers is not in the document]
Comment (maheshk114) [reply ql/src/java/org/apache/hadoop/hive/ql/security/authorization/StorageBasedAuthorizationProvider.java]: done

File: ql/src/java/org/apache/hadoop/hive/ql/security/authorization/StorageBasedAuthorizationProvider.java (hunk 2 above)
Comment (maheshk114) [reply ql/src/java/org/apache/hadoop/hive/ql/security/authorization/StorageBasedAuthorizationProvider.java]: done

**Pull request closed without merging**